src/arch/svm/SpokeUtils.ts

```diff
@@ -426,41 +427,146 @@ export function createTokenAccountsInstruction(
 }
 
 /**
- * @param mint Address of the token corresponding to the account being made.
- * @param amount Amount of the token to approve.
- * @param relayer Address of the relayer filling the deposit.
- * @param spokePool Address (program ID) of the SvmSpoke.
- * @returns A token approval instruction.
+ * Creates a fill instruction.
+ * @param signer - The signer of the transaction.
+ * @param solanaClient - The Solana client.
+ * @param fillInput - The fill input.
+ * @param tokenDecimals - The token decimals.
+ * @param createRecipientAtaIfNeeded - Whether to create a recipient token account.
+ * @returns The fill instruction.
  */
-export async function createApproveInstruction(
-  mint: SvmAddress,
-  amount: BigNumber,
-  relayer: SvmAddress,
-  spokePool: SvmAddress,
-  mintDecimals?: number
-) {
-  const [relayerTokenAccount, statePda] = await Promise.all([
-    getAssociatedTokenAddress(relayer, mint, TOKEN_PROGRAM_ADDRESS),
-    getStatePda(spokePool.toV2Address()),
-  ]);
+export const createFillInstruction = async (
```

Comment:
I moved these helpers to SpokeUtils. They were originally in test/utils, but I think they make more sense here. We can use them through wrapper functions like [getFillRelayTx](https://github.com/across-protocol/sdk/blob/9543adb024421815fc1bdb182f343a1ecbcad93b/src/relayFeeCalculator/chain-queries/svmQuery.ts#L131).